src/main/java/com/jfrog/xray/client/impl/XrayClient.java

```diff
@@ -20,20 +21,21 @@ static public Xray create(CloseableHttpClient preConfiguredClient, String url) {
     /**
      * Username, API key, and custom url
      */
-    static public Xray create(String url, String username, String password, String userAgent) {
+    static public Xray create(String url, String username, String password, String userAgent, ProxyConfig proxyConfig) {
         HttpBuilderBase configurator = new HttpBuilderBase() {
         };
         configurator.hostFromUrl(url)
                 .authentication(username, password, true)
                 .connectionTimeout(CONNECTION_TIMEOUT_MILLISECONDS)
                 .socketTimeout(CONNECTION_TIMEOUT_MILLISECONDS)
                 .userAgent(userAgent)
+                .proxy(proxyConfig)
                 .addRequestInterceptor(new PreemptiveAuthInterceptor());
 
         return new XrayImpl(configurator.build(), url);
     }
 
-    static public Xray create(String url, String username, String password) {
-        return create(url, username, password, DEFAULT_USER_AGENT);
+    static public Xray create(String url, String username, String password, ProxyConfig proxyConfig) {
```

Comment:
static public --> public static
There are a few other places in the code which require this change.